e2e/stack_monitoring_test.go

```diff
@@ -0,0 +1,614 @@
+			// whereas Metricbeat collection simply won't index it. So if we find kibana_settings.xpack.default_admin_email
+			// is null, we simply remove it
+			if docType == "kibana_settings" {
+				err := legacy.DeleteP("xpack.default_admin_email")
+				if err != nil {
+					return fmt.Errorf("Could not remove default_admin_email field")
+				}
+			}
+
+			return nil
+		}
+	case sm.Product == "logstash":
+		sm.Port = strconv.Itoa(9601)
+
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			pipelinesPath := "logstash_stats.pipelines"
+
+			if docType == "logstash_stats" {
+				legacyPipelines := legacy.Path(pipelinesPath)
+				metricbeatPipelines := metricbeat.Path(pipelinesPath)
+
+				legacyPipeline := legacyPipelines.Children()[0]
+				metricbeatPipeline := metricbeatPipelines.Children()[0]
+
+				legacyVertices := legacyPipeline.Path("vertices")
+				metricbeatVertices := metricbeatPipeline.Path("vertices")
+
+				// sort by vertex ID
+				log.Debug(legacyVertices)
+				log.Debug(metricbeatVertices)
+			}
+
+			return nil
+		}
+	case strings.HasSuffix(sm.Product, "beat"):
+		productIndexID = "beats"
+
+		// When Metricbeat monitors Filebeat, it encounters a different set of file IDs in
+		// `type:beats_stats` documents than when internal collection monitors Filebeat. However,
+		// we expect the _number_ of files being harvested by Filebeat in either case to match.
+		// If the numbers match we normalize the file lists in `type:beats_stats` docs collected
+		// by both methods so their parity comparison succeeds.
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			filesPath := "beats_stats.metrics.filebeat.harvester.files"
+
+			if docType == "beats_stats" {
+				legacyFiles := legacy.Path(filesPath)
+				metricbeatFiles := metricbeat.Path(filesPath)
+
+				legacyFilesCount := len(legacyFiles.Children())
+				metricbeatFilesCount := len(metricbeatFiles.Children())
+
+				if legacyFilesCount != metricbeatFilesCount {
+					return fmt.Errorf("The number of harvested files in legacy (%d) and metricbeat (%d) collection is different", legacyFilesCount, metricbeatFilesCount)
+				}
+
+				log.Debugf("The number of harvested files in legacy and metricbeat collection is the same: %d", legacyFilesCount)
+			}
+
+			return nil
+		}
+
+		sm.Port = strconv.Itoa(5066)
+
+		// look up configurations under workspace's configurations directory
+		dir, _ := os.Getwd()
+		env[sm.Product+"ConfigFile"] = path.Join(dir, "configurations", "parity-testing", sm.Product+".yml")
+
+		env["serviceName"] = sm.Product
+
+		if collectionMethod == "metricbeat" {
+			env["httpEnabled"] = "true"
+			env["httpPort"] = "5066"
+			env["xpackMonitoring"] = "false"
+		} else {
+			env["xpackMonitoring"] = "true"
+		}
+	default:
+		return fmt.Errorf("Product %s not supported", product)
+	}
+
+	sm.IndexName = ".monitoring-" + productIndexID + "-7"
```

Comment:
Hey @ycombinator, when running kibana or logstash, I do not see the related legacy monitoring index: `.monitoring-logstash`, so the program cannot query ES for the documents and die (expected). Any guess?

Reply:
In the case of Logstash, to configure "legacy" collection, you need to set `xpack.monitoring.enabled: true` in the Logstash node's configuration. Since we're running Logstash on Docker here, that can be done by setting the environment variable `XPACK_MONITORING_ENABLED=true` in Logstash's `docker-compose.yml`. 

For more info, see https://www.elastic.co/guide/en/logstash/current/docker-config.html#docker-env-config and https://www.elastic.co/guide/en/logstash/current/monitoring-internal-collection.html#monitoring-general-settings.

Reply:
Kibana is similar to Logstash. See https://www.elastic.co/guide/en/kibana/current/docker.html#environment-variable-config.